### ISSUE TITLE
Refactor ReflectiveConfigOptionLoader

### ISSUE THREAD

**Issue by dalaro** [thinkaurelius/titan#711]
I want to make several changes to ReflectiveConfigOptionLoader.
- Add a way to disable it entirely
- Move ReflectiveConfigOptionLoader's logic into the ConfigOption/ConfigNamespace/ConfigElement implementation where it as actually needed, so that it can't be accidentally bypassed through unusual construction of TitanGraph or its Backend classes (only by deliberately disabling it as mentioned in the previous item)
- Force Titan's standard config options to preload without resorting to a costly classpath walk.  The common case involves only ConfigOptions that we know about at compile time, and we should optimize that case to run in milliseconds instead of seconds.  I haven't decided exactly how I want the details of this last optimization to look, but we have plenty of maneuvering room to make it faster.

**Comment by mbroecheler** [comment]
It would be good if we could reduce the time it takes to startup a TitanGraph due to the classpath loading.

**Comment by dalaro** [comment]
I rewrote this class in 69ca65b21796f5639b3ef77a71b42e91a3d12212 and 0b2620d9aec9292a4e79f53291dbc50689b7b130.  Here's a summary of changes.
- Classloader configuration: the sequence of classloaders consulted when loading options is now user-configurable to support environments with unusual classloader hierarchies.  Use of the "java.class.path" system property in Reflections has been replaced with use of this configurable classloader chain (#712).
- Disable switch: the reflective config option loader can be completely disabled by setting the system property `titan.load.cfg.opts` or environment variable `TITAN_LOAD_CFG_OPTS` to `false`, or by calling `ReflectiveConfigOptionLoader.INSTANCE.setEnabled(false)`.  This puts responsibility for executing config option static initializers completely on the client.
- Fewer call sites: the reflective config option loader is used only in `ConfigNamespace.getChild(String)` and in `ConfigurationPrinter` (the latter generates the config tables in the manual and isn't used outside of documentation generation).  Whenever `getChild` is about to return null, it invokes the fast path, tries to resolve the child again, invokes the slow path if still null, tries to resolve the child again, and finally returns the result.  The fast and slow paths generally only execute once each, although they may execute more than once each through concurrent threads.  There is no synchronization, but there is one volatile read per invocation.  This obviates the calls in TitanFactory.open and TitanFactory.Builder and reduces the likelihood of unintentionally bypassing the option loader, though it can still be intentionally bypassed as described in the previous point.
- Speed: it's now optimized for the case where only standard Titan ConfigOptions are present (no third-party options).  This case runs in about 60~80 ms right now, compared with several seconds before.  The implementation still falls back on the general slow path if `ConfigNamespace.getChild(String)` is about to return null, as described in the previous point.
